pocs/core.py

```diff
@@ -434,6 +437,65 @@ def sleep(self, delay=2.5, with_status=True):
         if delay > 0.0:
             time.sleep(delay)
 
+    def wait_for_events(self,
+                        events,
+                        timeout,
+                        sleep_delay=1,
+                        status_interval=10,
+                        msg_interval=30,
+                        event_type='generic'):
+        """Wait for event(s) to be set.
+
+        This method will wait for a maximum of `timeout` seconds for all of the
+        `events` to complete. The loop will `check_messages` each time in case of
```

Comment:
Will check at least every `sleep_delay` seconds for the events to be done, and also for interrupts and bad weather. Will log debug messages approximately every `status_interval` seconds, and will output status messages approximately every `status_interval` seconds.

Reply:
Thanks, I think I was getting lazy. :)